rust/agents/relayer/src/msg/metadata/base.rs

```diff
@@ -1,39 +1,41 @@
 use async_trait::async_trait;
 use num_derive::FromPrimitive;
 use num_traits::FromPrimitive;
-use std::fmt::Debug;
 use std::str::FromStr;
 use std::sync::Arc;
-use std::{collections::HashMap, ops::Deref};
+use std::{collections::HashMap, fmt::Debug};
 
 use derive_new::new;
 use eyre::Context;
 use tokio::sync::RwLock;
-use tracing::{debug, info, instrument, warn};
+use tracing::{debug, instrument, warn};
 
 use hyperlane_base::{
     ChainSetup, CheckpointSyncer, CheckpointSyncerConf, CoreMetrics, MultisigCheckpointSyncer,
 };
-use hyperlane_core::{HyperlaneMessage, MultisigIsm, ValidatorAnnounce, H160, H256};
+use hyperlane_core::{HyperlaneMessage, ValidatorAnnounce, H160, H256};
 
 use crate::merkle_tree_builder::MerkleTreeBuilder;
+use crate::msg::metadata::{MultisigIsmMetadataBuilder, RoutingIsmMetadataBuilder};
 
 #[derive(Debug, thiserror::Error)]
 pub enum MetadataBuilderError {
     #[error("Unknown or invalid module type ({0})")]
     UnsupportedModuleType(u8),
+    #[error("Exceeded max depth when building metadata ({0})")]
+    MaxDepthExceeded(u32),
 }
 
-#[derive(FromPrimitive)]
+#[derive(FromPrimitive, Clone, Debug)]
 pub enum SupportedIsmTypes {
-    // Routing = 1,
+    Routing = 1,
     // Aggregation = 2,
     LegacyMultisig = 3,
-    // Multisig = 4,
+    Multisig = 4,
```

Comment:
any way we can derive this from the contract?

Reply:
It's not in the ABI, so I doubt it